typescript/cli/src/tests/warp-apply.e2e-test.ts

```diff
@@ -0,0 +1,132 @@
+import { expect } from 'chai';
+import { Wallet } from 'ethers';
+import { beforeEach } from 'mocha';
+
+import { ChainAddresses } from '@hyperlane-xyz/registry';
+import { TokenRouterConfig, TokenType } from '@hyperlane-xyz/sdk';
+
+import {
+  ANVIL_KEY,
+  REGISTRY_PATH,
+  deployOrUseExistingCore,
+  extendWarpConfig,
+  getChainId,
+  updateOwner,
+} from './commands/helpers.js';
+import { hyperlaneWarpDeploy, readWarpConfig } from './commands/warp.js';
+
+/// To run: 1) start 2 anvils, 2) yarn run tsx tests/warp.zs-test.ts inside of cli/
+const CHAIN_NAME_1 = 'anvil1';
+const CHAIN_NAME_2 = 'anvil2';
+
+const BURN_ADDRESS = '0x0000000000000000000000000000000000000001';
+const EXAMPLES_PATH = './examples';
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config.yaml`;
+const WARP_CONFIG_PATH = `${EXAMPLES_PATH}/warp-route-deployment.yaml`;
+const WARP_CORE_CONFIG_PATH_1 = `${REGISTRY_PATH}/deployments/warp_routes/ETH/anvil1-config.yaml`;
+
+describe('WarpApply e2e tests', async function () {
+  this.timeout(0); // No limit timeout since these tests can take a while
```

Comment:
I believe you can hook this `.timeout` call to the back of the describe fn, which looks more natural IMHO

Reply:
will fix in a separate PR